### PR TITLE
[API] Refactored ISarosSession

### PR DESCRIPTION
**Pull request by pdler**
ISarosSession/SarosSession: 
The interface ISarosSession was adjusted to IReferencePoints. So every class that contains a SarosSession object,  has to give referencePoints instead of the projects. Every class that wants to add resources to the session, has to put  the mapping of a referencePoint to his projects in the referencePointManager.  Because of refactoring the SarosSession interface, many classes are affected but the changes are minimal (mainly using IReferencePoint instead of core.IProject as parameter)

The SarosSession itself was not adjusted completely to reference points because the activities still depends on projects.

In issue #177 is a detailed description, what an IReferencePoint is, for what we need this and which steps planned for the future commits

### CODE REVIEW

File: de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/negotiation/ProjectNegotiation.java [hunk 1]
```diff
@@ -71,6 +73,8 @@
 
     protected TransferType transferType;
 
+    protected IReferencePointManager referencePointManager;
+
     public ProjectNegotiation(final String id, final JID peer,
```
Comment (saros-infrastructure) [inline de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/negotiation/ProjectNegotiation.java]: Code Smell Code Smell: Constructor has 10 parameters, which is greater than 7 authorized. ([squid:S00107](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AS00107))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=258&issues=AWbe6KEnSgN4uCclR4y6&open=AWbe6KEnSgN4uCclR4y6)

[ISSUE_KEY]: AWbe6KEnSgN4uCclR4y6

File: de.fu_berlin.inf.dpp/src/de/fu_berlin/inf/dpp/ui/eventhandler/SessionStatusRequestHandler.java [hunk 2]
```diff
@@ -82,16 +87,17 @@ private void handleStatusRequest(JID from) {
...
+            .getReferencePoints());
         int i = 0;
         int numOfProjects = projects.size();
 
         for (IProject project : projects) {
             description += project.getName();
 
-            if (!session.isCompletelyShared(project))
+            if (!session.isCompletelyShared(project.getReferencePoint()))
                 description += " (partial)";
```
Comment (saros-infrastructure) [inline de.fu_berlin.inf.dpp/src/de/fu_berlin/inf/dpp/ui/eventhandler/SessionStatusRequestHandler.java]: Code Smell Code Smell: Use a StringBuilder instead. ([squid:S1643](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AS1643))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=258&issues=AWbe6KHGSgN4uCclR40F&open=AWbe6KHGSgN4uCclR40F)

[ISSUE_KEY]: AWbe6KHGSgN4uCclR40F

File: de.fu_berlin.inf.dpp/src/de/fu_berlin/inf/dpp/ui/eventhandler/SessionStatusRequestHandler.java [hunk 3]
```diff
@@ -82,16 +87,17 @@ private void handleStatusRequest(JID from) {
...
+        String description = "ReferencePoints: ";
 
-        Set<IProject> projects = session.getProjects();
+        Set<IProject> projects = referencePointManager.getProjects(session
+            .getReferencePoints());
         int i = 0;
         int numOfProjects = projects.size();
 
         for (IProject project : projects) {
             description += project.getName();
```
Comment (saros-infrastructure) [inline de.fu_berlin.inf.dpp/src/de/fu_berlin/inf/dpp/ui/eventhandler/SessionStatusRequestHandler.java]: Code Smell Code Smell: Use a StringBuilder instead. ([squid:S1643](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AS1643))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=258&issues=AWbe6KHGSgN4uCclR40E&open=AWbe6KHGSgN4uCclR40E)

[ISSUE_KEY]: AWbe6KHGSgN4uCclR40E

File: de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/negotiation/stream/OutgoingStreamProtocol.java [hunk 4]
```diff
@@ -82,7 +82,8 @@ public void streamFile(SPath file) throws IOException,
     }
 
     private void writeHeader(SPath file, long fileSize) throws IOException {
-        String projectID = session.getProjectID(file.getProject());
+        String projectID = session.getReferencePointID(file.getProject()
```
[the remark this answers is not in the document]
Comment (pdler) [reply de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/negotiation/stream/OutgoingStreamProtocol.java]: the parameter 'file' is a SPath object. SPath will get the method getReferencePoint() later in a commit. I renamed 'file' to 'path' instead.

File: de.fu_berlin.inf.dpp/src/de/fu_berlin/inf/dpp/ui/eventhandler/SessionStatusRequestHandler.java (hunk 3 above)
Comment (saros-infrastructure) [inline de.fu_berlin.inf.dpp/src/de/fu_berlin/inf/dpp/ui/eventhandler/SessionStatusRequestHandler.java]: Code Smell Code Smell: Use a StringBuilder instead. ([squid:S1643](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AS1643))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=258&issues=AWbe6KHGSgN4uCclR40E&open=AWbe6KHGSgN4uCclR40E)

[ISSUE_KEY]: AWbe6KHGSgN4uCclR40E

File: de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/negotiation/ProjectNegotiation.java (hunk 1 above)
Comment (saros-infrastructure) [inline de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/negotiation/ProjectNegotiation.java]: Code Smell Code Smell: Constructor has 10 parameters, which is greater than 7 authorized. ([squid:S00107](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AS00107))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=258&issues=AWbe6KEnSgN4uCclR4y6&open=AWbe6KEnSgN4uCclR4y6)

[ISSUE_KEY]: AWbe6KEnSgN4uCclR4y6

File: de.fu_berlin.inf.dpp/src/de/fu_berlin/inf/dpp/ui/eventhandler/SessionStatusRequestHandler.java (hunk 2 above)
Comment (saros-infrastructure) [inline de.fu_berlin.inf.dpp/src/de/fu_berlin/inf/dpp/ui/eventhandler/SessionStatusRequestHandler.java]: Code Smell Code Smell: Use a StringBuilder instead. ([squid:S1643](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AS1643))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=258&issues=AWbe6KHGSgN4uCclR40F&open=AWbe6KHGSgN4uCclR40F)

[ISSUE_KEY]: AWbe6KHGSgN4uCclR40F

**Comment by tobous** [comment]
Deprecated.

**Pull request closed without merging**